lib/pbench/test/functional/server/test_put.py

```diff
@@ -85,27 +85,24 @@ def test_index_all(self, server_client: PbenchServerClient, login_user):
 
         # For each dataset we find, poll the state until it reaches Indexed
         # state, or until we time out.
-        start = time.time()
+        now = start = time.time()
         timeout = start + (60.0 * 10.0)
 
-        while True:
-            now = time.time()
+        while not_indexed:
             print(f"[{now - start:0.2f}] Checking ...")
             not_indexed, indexed = TestPut.check_indexed(server_client, not_indexed)
             for dataset in indexed:
                 count += 1
-                print(
-                    "    Indexed ",
-                    dataset.metadata["server.tarball-path"],
-                )
-            if not not_indexed:
+                print("    Indexed ", dataset.metadata["server.tarball-path"])
+            if not not_indexed or now >= timeout:
                 break
-            if time.time() < timeout:
-                time.sleep(30.0)  # sleep for half a minute
-            else:
-                print("Exceeded timeout (10 minutes), still not indexed:")
-                for dataset in not_indexed:
-                    print(dataset.metadata["server.tarball-path"])
+            time.sleep(30.0)  # sleep for half a minute
+            now = time.time()
```

Comment:
I think it works better to position the assignment of `now` at the top of the loop (which relieves us from having to pre-assign it at line 88, with the double-assignment thing which I dislike).

(I'm going to pre-resolve this....)